### PR TITLE
Oozelings can now extinguish people with hugs!

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

It's quite self-explanatory. Oozelings can now extinguish fires on people by hugging them. If they're fireproof, it'd make sense they could smother a fire on someone else with a tight hug.

Just to prevent an oozeling from spam clicking a bunch of people to constantly put fires out, there's a 1 second do_after when hugging someone on fire as an ooze.

## Why It's Good For The Game

Why not? It kinda just makes sense.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/65794972/232264098-178fcdbe-03a4-4dd6-b583-60b8a897d83e.mp4

</details>

## Changelog
:cl:
add: Oozelings can now hug flaming people to extinguish them!
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
This is a species change PR, should probably get maintainer approval for it.

**Comment by Absolucy** [comment]
@CydiaLamiales Oh, sorry. Uh, who do I ask?

**Comment by CydiaLamiales** [comment]
Any maintainer

**Comment by Absolucy** [comment]
Closing until I get maintainer permission, then.

**Comment by Absolucy** [comment]
image
i guess this means i can reopen this

**Comment by PowerfulBacon** [comment]
Maintainer approval is required for all species changes. Kat is refering to the thing in the design notes which is saying that we can close felinid/moth PRs if we feel like it without a specific reason rather than the bit about species change.
I'll give my approval for this, I think it should work like pressing B, where if someone is on fire a lot then it won't extinguish them, only reduce the flames a bit.

**Comment by Absolucy** [comment]
@PowerfulBacon how many firestacks do you think it should take off each time? i'm not too familiar with how many firestacks is a lot.

**Comment by PowerfulBacon** [comment]
Check what stop, drop and roll does and copy the same amount as that

**Comment by Absolucy** [comment]
@PowerfulBacon done

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species_types/oozelings.dm]: It should work on any bodypart selected imo, may be awkward to have to select chest every time

Comment (Absolucy) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm]: fixed

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm
Comment (Rukofamicom) [inline code/modules/mob/living/carbon/human/species_types/oozelings.dm:196]: Won't this message display even if the oozeling is the one that moves or otherwise disengages the hug? I feel like this should be worded differently to reflect that possibility

**Review by Rukofamicom — CHANGES_REQUESTED** [review]

**Comment by Rukofamicom** [comment]
For what it's worth I like this 👍

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm
Comment (Absolucy) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm:196]: Hm, how would you word it?

Comment (Rukofamicom) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm:196]: I've been staring at this for ten minutes and not really had any good solution come up yet. 🤔
I might have to get back to you on this, I'm stumped right now.

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm
Comment (itsmeow) [inline code/modules/mob/living/carbon/human/species_types/oozelings.dm]: This to_chat is unnecessary

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm
Comment (itsmeow) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm:196]: Yeah it's good enough and I can't think of anything better

**Review by itsmeow — APPROVED** [review]
Removing firestacks 👍🏻 
Making species more unique 👍🏻 
Giving species actual advantages 👍🏻

**Pull request merged**
